Fix travis build and simplify config

lostapathy opened the pull request:
I left this as two commits to clarify what I was doing.

The first recognizes that USE_RVM was set on every build, so I removed the conditional case for when it wasn't, and then removed the env variable to simplify the script.

The second commit updates to the latest ruby versions and uses simpler travis syntax to define the build matrix.

lostapathy commented:
I will put the OSX builds in another PR later, apologies for the noise.

lostapathy commented:
@samaaron any interest in this?  Happy to update it to trunk and work on OSX builds if there is, but if not will just close this.

samaaron commented:
Sorry for the slow response. I’ve been free lots lot to try and drum up new suppPet for continued development now that Raspberry Pi have stopped supporting me financially. 

I am interested in this - especially if you can build for macOS :)

lostapathy commented:
Good luck on sponsorship @samaaron 
I updated this patch to apply cleanly again and squashed the history.  Once this is merged I'll start on the osx builds.

samaaron commented:
Argh, sorry I've left this too long again. I added support for testing Ruby 2.5 to the travis file recently which I guess caused this recent conflict.

Could we try once more? Many apologies.

lostapathy commented:
Sure thing @samaaron.  I fixed the conflict and bumped the other ruby versions, this should be good to go.

samaaron commented:
Wonderful, thanks :-)
